src/module.c

```diff
@@ -5853,7 +5853,17 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
     }
 
     if (flags & REDISMODULE_ARGV_RESPECT_DENY_OOM) {
-        if (cmd->flags & CMD_DENYOOM) {
+        uint64_t flags = cmd->flags;
```

Comment:
i think we need to do all that further above, so we can also affect the CMD_WRITE and CMD_STALE checks.